server/api/app/Applications/User/Controllers/UserController.php

```diff
@@ -156,4 +158,42 @@ public function updateMyProfile(MyProfile $request)
     {
         $this->userService->updateMyProfile($request);
     }
+
+    /**
+     * Handle the avatar upload for the authenticated user.
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function uploadAvatar(Request $request): JsonResponse
+    {
+        try {
+            $userId = Route::current()->parameter('id');
+            $authenticatedUser = Auth::user();
+
+            $userDTO = $this->userService->uploadAvatar($userId, $request, $authenticatedUser);
+
+            return response()->json($userDTO, 200);
+        } catch (ValidationException $e) {
+            return response()->json([
+                'message' => 'Validation error.',
+                'errors' => $e->errors(),
+            ], 422);
+        } catch (\Symfony\Component\HttpKernel\Exception\NotFoundHttpException $e) {
+            return response()->json([
+                'message' => 'User not found.',
+            ], 404);
+        } catch (AuthorizationException $e) {
+            return response()->json([
+                'message' => $e->getMessage(),
+            ], 403);
+        } catch (\Exception $e) {
+            // Log unexpected errors and return a generic error response
+            \Log::error('Error uploading avatar: ' . $e->getMessage(), ['exception' => $e]);
+
+            return response()->json([
+                'message' => 'An error occurred while uploading the avatar. Please try again later.',
```

Comment:
This is probably correct but I think it was possible to globally register a custom exception renderer. https://laravel.com/docs/11.x/errors#renderable-exceptions
Basically if 500 return your custom message "An error occurred while uploading the avatar. Please try again later" ....